packages/common-ui-web/src/components/VehicleDynamicWireframe/VehicleDynamicWireframe.tsx

```diff
@@ -0,0 +1,45 @@
+import { useMonkTheme } from '@monkvision/common';
+import { wireFrame } from '@monkvision/sights';
+import { PartSelectionOrientation, VehicleModel, VehiclePart } from '@monkvision/types';
+import { useMemo } from 'react';
+import { DynamicSVG } from '../DynamicSVG';
+import { usePartsOverlay } from './hooks';
+
+export interface VehicleDynamicWireframeProps {
+  vehicleModel: VehicleModel;
+  orientation?: PartSelectionOrientation;
+  parts: VehiclePart[];
+  /**
+   * emit when change in the selected parts.
+   */
+  onPartsSelected?: (parts: VehiclePart[]) => void;
+}
+
+/**
+ * Component that displays a dynamic wireframe of a vehicle,
+ * allowing the user to select parts of the vehicle.
+ */
+export function VehicleDynamicWireframe({
```

Comment:
Okay so I think there is a bit a bad pattern here. Let me explain. Here, you decided to handle the part selection logic from within the `VehicleDynamicWireframe`. This should not be the case. This component should just be a "UI" component, that displays the required wireframe, fire events when interacted with, and be customizable via props (custom styles for parts etc.).

This means that instead of what you have here, the `VehicleDynamicWireframe` should accept a prop like this : 

```typescript
export interface VehicleDynamicWireframeProps {
  ...
  getPartAttributes?: (part: VehiclePart) => SVGProps<SVGElement>,
}
```

This would allow developers to customize the look and feel of the vehicle parts while using this component.

Basically :

- In the `VehiclePartSelection` component, you should have all the logic to handle :
  - Currently selected vehicle parts
  - Toggle vehicle part selection when `onPartClick` events are fired
  - Create custom styles for each part when they are selected and pass it to the `VehicleDynamicWireframe` via the `getPartAttributes` prop

Since this is a big change in your code, I'm going to stop reviewing the PR now. This is the last comment I add to your PR. Once you have fixed every comment (including this one), add thumbs ups, ping me and I'll finish the review.